Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/SubmissionCardState.kt

```diff
@@ -17,16 +18,33 @@ import de.rki.coronawarnapp.util.NetworkRequestWrapper.Companion.withSuccess
 
 data class SubmissionCardState(
     val deviceUiState: NetworkRequestWrapper<DeviceUIState, Throwable>,
-    val isDeviceRegistered: Boolean
+    val isDeviceRegistered: Boolean,
+    val submissionRepository: SubmissionRepository
 ) {
 
-    fun isRiskCardVisible(): Boolean =
-        deviceUiState.withSuccess(true) {
-            when (it) {
-                PAIRED_POSITIVE, PAIRED_POSITIVE_TELETAN, SUBMITTED_FINAL -> false
-                else -> true
+    private val testViewed = submissionRepository.hasViewedTestResult
+
+    fun isRiskCardVisible(): Boolean {
```

Comment:
We could expose LiveData here to make the view reactive. This would work well with the flow from the repository.